src/components/adminWaiverEdit.vue

```diff
@@ -0,0 +1,395 @@
+      inputLabelTableLarge,
+      inputLabelOnly,
+      waiverReadOnly
+    },
+    data() {
+      return {
+        waiver: {
+          title: '',
+          header: [],
+          items: [],
+          footer: '',
+
+          initials: [],
+          signature: '[MY FULL NAME]',
+          signed_date: '',
+        },
+
+        waiver_labels: {
+          title: "Title",
+          header: "Header Text",
+          items: "Legal Items",
+          item_order: "Order #",
+          item_title: "Title",
+          item_initials: "Initials Required?",
+          item_text: "Text",
+          footer: "Footer"
+        },
+
+        waiver_help: {
+          item_order: "This will define the order the items will appear on the waiver."
+        },
+
+        waiver_validators: {
+          title_min: 2,
+          header_min: 2,
+          item_order_min: 0,
+          item_text_min: 10,
+          item_title_min: 2,
+          footer_min: 2,
+        },
+
+        invalid: {
+          title: false,
+          header: [],
+          items: [],
+          footer: false,
+
+          initials: [],
+          signature: true,
+          signed_date: true,
+        },
+
+        invalid_item: {
+          'order': false,
+          'title': false,
+          'initials': false,
+          'text': false
+        },
+
+        type: '',
+        type_cap: '',
+
+        preview_height: '',
+        preview_btn_styles: {},
+
+        headers_have_errors: false,
+        items_have_errors: false,
+      };
+    },
+    methods: {
+      getWaiver: function(type) {
+        // get waiver text from database storage
+        let resource_type = "waiver_" + type;
+        this.$store.dispatch('getWaiverResources', {
+          resource: resource_type
+        })
+        .then(data => {
+          this.waiver.title = data['title'];
+          this.waiver.header = data['headers'];
+          this.waiver.items = data['items'];
+          this.waiver.footer = data['footer'];
+
+          // init invalid list for each header
+          this.waiver.header.forEach(h => {
+            this.invalid.header.push(false);
+          });
+
+          // init invalid list for each item
+          this.waiver.items.forEach(i => {
+            this.invalid.items.push(this.invalid_item);
+          });
+          setTimeout(x => {this.matchHeight()}, 2000);
+        })
+      },
+
+      matchHeight() {
+        let preview_height = this.$refs.updateWaiver.scrollHeight - 50;
+        this.preview_btn_styles = {'height': preview_height + 'px'};
+      },
+
+      scrollToPreview() {
+        let element = '#waiver_preview';
+        let duration = 2000;
+        var VueScrollTo = require('vue-scrollto');
+        VueScrollTo.scrollTo(element, duration)
+      },
+
+      deleteItem(item_idx) {
+        // remove the item from the list
+        this.waiver.items.splice(item_idx, 1);
+      },
+
+      addItem() {
+        let count = this.waiver.items.length;
```

Comment:
need event.preventDefault() here, so it doesn't refresh

Reply:
great catch. done.

Reply:
in vue it should be set as
v-on:click.stop.prevent